File: sdk/python/test_vines.py
import os
import sys
from ScyKernel import ScyKernel

def run_test():
    db_dir = "vines_images"
    path_ppm = os.path.join(db_dir, "py_vine.ppm")
    path_png = os.path.join(db_dir, "py_vine.png")
    test_key = "User"
    test_value = "Amanda"
    #test_value = "<bos> " + ("STRESS_TEST_DATA_" * 280) + " <eos>"
    password = "ScyWeb_Global_Secret_2026"
    if not os.path.exists(db_dir):
        os.makedirs(db_dir)
    scy = ScyKernel(password, path_ppm)
    scy.create_ppm_db(path_ppm)
    scy.sync_png(path_png, "load")
    scy.put_to_ppm(test_key, test_value, password)
    scy.put_to_png(test_key, test_value, password)
    scy.sync_png(path_png, "commit")
    scy.sync_png(path_png, "load")
    result_ppm = scy.get_from_ppm(test_key, password)
    result_png = scy.get_from_png(test_key, password)
    if result_ppm == test_value and result_png == test_value:
        validation = "Valid" if scy.validate_db(path_ppm) else "Invalid"
        print(f"✅ Python KV Parity: SUCCESS (Recovered: {result_ppm})")
        print(f"🧩 PPM is: {validation}")
        print(f"📏 Size of Image DB: {scy.get_file_size(path_png)} bytes")
        parity_configs = [
            ("C++", "../cpp/vines_images/cpp_vine.png"),
            ("Go", "../go/scykernel/vines_images/go_vine.png"),
            ("Java", "../java/vines_images/java_vine.png"),
            ("Node", "../javascript/vines_images/node_vine.png"),
            ("Kotlin", "../kotlin/vines_images/kt_vine.png"),
            ("PHP", "../php/vines_images/php_vine.png"),
            ("Python", "../python/vines_images/py_vine.png"),
            ("React Native", "../react-native/vines_images/rn_vine.png"),
            ("Rust", "../rust/vines_images/rust_vine.png"),
            ("Swift", "../swift/vines_images/swift_vine.png")
        ]
        for lang, l_path in parity_configs:
            if os.path.exists(l_path):
                if scy.sync_png(l_path, "load"):
                    res = scy.get_from_png(test_key, password)
                    if res == test_value:
                        print(f"✅ Python to {lang} Parity: SUCCESS (Recovered: {res})")
                    else:
                        print(f"❌ Python to {lang} Parity: FAIL")
        sys.exit(0)
    else:
        print("❌ Python KV Parity: FAIL")
        print(f"Expected: {test_value}, Got PPM: [{result_ppm}], PNG: [{result_png}]")
        scy.delete_db(path_ppm)
        scy.delete_db(path_png)
        sys.exit(1)

if __name__ == "__main__":
    run_test()

File: sdk/java/test_vines.java
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Map;
import java.util.LinkedHashMap;

public class test_vines {
    public static void main(String[] args) {
        final String dir = "vines_images";
        final String pathPPM = dir + "/java_vine.ppm";
        final String pathPNG = dir + "/java_vine.png";
        final String testKey = "User";
        final String testValue = "Amanda";
        final String password = "ScyWeb_Global_Secret_2026";

        try {
            // Ensure the local folder exists
            Files.createDirectories(Paths.get(dir));

            // Instantiate ScyKernel (Constructor: password, default path)
            ScyKernel scy = new ScyKernel(password, pathPPM);

            // Creating the test DBs
            scy.createPPM_DB(pathPPM);
            scy.syncPNG(pathPNG, "load");

            // Test both PPM and PNG DBs
            scy.putToPPM(testKey, testValue, password);
            scy.putToPNG(testKey, testValue, password);

            // sync changes and refresh
            scy.syncPNG(pathPNG, "commit");
            scy.syncPNG(pathPNG, "load");

            // Retrieve the results from both DBs
            String resultPPM = scy.getFromPPM(testKey, password);
            String resultPNG = scy.getFromPNG(testKey, password);

            // Output Comparison
            if (testValue.equals(resultPPM) && testValue.equals(resultPNG)) {
                String validationStatus = scy.validateDB(pathPPM) ? "Valid" : "Invalid";
                System.out.printf("✅ Java KV Parity: SUCCESS (Recovered: %s)%n", resultPPM);
                System.out.printf("🧩 PPM is: %s%n", validationStatus);
                long pngSize = scy.getFileSize(pathPNG);
                System.out.printf("📏 Size of Image DB: %d bytes%n", pngSize);
                //scy.deleteDB(pathPPM);
                //scy.deleteDB(pathPNG);
                Map<String, String> parityConfigs = new LinkedHashMap<>();
                parityConfigs.put("C++", "../cpp/vines_images/cpp_vine.png");
                parityConfigs.put("Go", "../go/scykernel/vines_images/go_vine.png");
                parityConfigs.put("Java", "../java/vines_images/java_vine.png");
                parityConfigs.put("Node", "../javascript/vines_images/node_vine.png");
                parityConfigs.put("Kotlin", "../kotlin/vines_images/kt_vine.png");
                parityConfigs.put("PHP", "../php/vines_images/php_vine.png");
                parityConfigs.put("Python", "../python/vines_images/py_vine.png");
                parityConfigs.put("React Native", "../react-native/vines_images/rn_vine.png");
                parityConfigs.put("Rust", "../rust/vines_images/rust_vine.png");
                parityConfigs.put("Swift", "../swift/vines_images/swift_vine.png");
                for (Map.Entry<String, String> entry : parityConfigs.entrySet()) {
                    String lang = entry.getKey();
                    String lPath = entry.getValue();
                    if (new File(lPath).exists()) {
                        ScyKernel scyCheck = new ScyKernel(password, lPath);
                        if (scyCheck.syncPNG(lPath, "load")) {
                            String res = scyCheck.getFromPNG(testKey, password);
                            if (res.equals(testValue)) {
                                System.out.println("✅ Java to " + lang + " Parity: SUCCESS (Recovered: " + res + ")");
                            } else {
                                System.out.println("❌ Java to " + lang + " Parity: FAIL");
                            }
                        }
                    }
                }
                System.exit(0);
            } else {
                System.out.println("❌ Java KV Parity: FAIL");
                System.out.printf("Expected: %s%n", testValue);
                System.out.printf("Got PPM: [%s]%n", resultPPM);
                System.out.printf("Got PNG: [%s]%n", resultPNG);
                scy.deleteDB(pathPPM);
                scy.deleteDB(pathPNG);
                System.exit(1);
            }

        } catch (Exception e) {
            System.err.println("❌ Java SDK Critical Error: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}